code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -163,6 +164,84 @@ func (ssc *StorageSmartContract) GetBlobbersHandler(
 	return sns, nil
 }
 
+// GetAllocationBlobbersHandler returns list of all blobbers alive (e.g. excluding
+// blobbers with zero capacity).
+func (ssc *StorageSmartContract) GetAllocationBlobbersHandler(
+	ctx context.Context,
+	params url.Values, balances cstate.StateContextI) (interface{}, error) {
+
+	var creationDate = common.Timestamp(time.Now().Unix())
+
+	allocData := params.Get("allocation_data")
+	var request newAllocationRequest
+	if err := request.decode([]byte(allocData)); err != nil {
+		return "", common.NewErrInternal("can't decode allocation request", err.Error())
+	}
+
+	var sa = request.storageAllocation()
+	if balances.GetEventDB() == nil {
+		return nil, errors.New("events db is not initialised")
+	}
+
+	blobbers, err := getBlobbersFromQuery(params, balances.GetEventDB())
+	if err != nil || len(blobbers) == 0 {
+		return nil, fmt.Errorf("no blobbers found %v", err)
+	}
+
+	var sns StorageNodes
+	for _, blobber := range blobbers {
+		sn, err := blobberTableToStorageNode(blobber)
+		if err != nil {
+			return ssc.GetBlobbersHandlerDeprecated(ctx, params, balances)
+		}
+		sns.Nodes.add(&sn)
+	}
+
+	var size = sa.DataShards + sa.ParityShards
+
+	var bSize = (sa.Size + int64(size-1)) / int64(size)
+
+	var list = sa.filterBlobbers(sns.Nodes.copy(), creationDate,
+		bSize, filterHealthyBlobbers(creationDate),
+		ssc.filterBlobbersByFreeSpace(creationDate, bSize, balances))
+
+	if len(list) < size {
+		return nil, errors.New("not enough blobbers to honor the allocation")
+	}
+
+	return list, nil
+}
+
+func getBlobbersFromQuery(params url.Values, eventsDB *event.EventDb) ([]event.Blobber, error) {
+	var blobbers []event.Blobber
+	dbStore := eventsDB.Store.Get()
```

Comment:
We should not be using event database objects directly in the endpoints. These should be contained in the event database code.

The problem is if we later wish to switch to a more efficient database solution than PostgreSQL/gorm. It is better if we just need to support an event database API, rather than have bits spread around everywhere.

Reply:
The `url.Values` object is for the boundary layer, I don't think it belongs in the event database, which does not need to know how clients send information to 0Chain.

I suggest reading the params into strings in GetAllocationBlobbersHandler and then pass them into the event databse,